.github/workflows/ci.yml

```diff
@@ -40,3 +40,30 @@ jobs:
         run: yarn run lint
       - name: Prettier
         run: yarn run prettier-check
+
+  release:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.repository == 'relayjs/eslint-plugin-relay'
+    needs: [build, lint]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16.x
+          cache: 'yarn'
+      - name: Build latest (main) version
+        if: github.ref == 'refs/heads/main'
+        run: yarn version --no-git-tag-version --new-version 0.0.0-main-${GITHUB_SHA}
+      - name: Build release version
+        if: github.ref_type == 'tag' && startsWith(github.ref_name, 'v')
+        run: yarn version --no-git-tag-version --new-version ${GITHUB_REF_NAME:1}
```

Comment:
But if this push turns out to be for a tag, the version will get updated to follow the version substring of the tag name.